### PR TITLE
pybind/rgwfile: fix incompatible with ubuntu cython

### PR DESCRIPTION
**Pull request by yuyuyu101**
Signed-off-by: Haomai Wang <haomai@xsky.com>

### CODE REVIEW

File: src/pybind/rgw/rgw_file.pyx [hunk 1]
```diff
@@ -99,7 +99,8 @@ cdef extern from "rados/rgw_file.h" nogil:
         uint64_t  f_files    # inodes
         uint64_t  f_ffree    # free inodes
         uint64_t  f_favail   # free inodes for unprivileged users
-        uint64_t  f_fsid[2]  # /* file system ID
+        uint64_t  f_fsid     # /* file system ID
+        uint64_t  f_fsid2    # /* file system ID 2
```
Comment (tchaikov) [inline src/pybind/rgw/rgw_file.pyx]: probably, we can return `(statbuf.f_fsid, statbuf.f_fsid2)` in `LibRGWFS. statfs`?

**Comment by yuyuyu101** [comment]
done

File: src/pybind/rgw/rgw_file.pyx [hunk 2]
```diff
@@ -406,6 +407,7 @@ cdef class LibRGWFS(object):
                 'f_ffree': statbuf.f_ffree,
                 'f_favail': statbuf.f_favail,
                 'f_fsid': statbuf.f_fsid,
+                'f_fsid2': statbuf.f_fsid2,
```
Comment (tchaikov) [inline src/pybind/rgw/rgw_file.pyx]: how about 
```python
'f_fsid': (statbuf.f_fsid, statbuf.f_fsid2),
```
?

Comment (tchaikov) [inline src/pybind/rgw/rgw_file.pyx]: pushed the suggested change to ceph/ceph:wip-11820, so gitbuilders can test the change.

**Comment by xiexingguo** [comment]
Still does not compile:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/rgw/rgw_file.c: In function ‘__pyx_pf_8rgw_file_8LibRGWFS_18statfs’:
/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/rgw/rgw_file.c:3243:3: warning: passing argument 1 of ‘__Pyx_PyInt_From_uint64_t’ makes integer from pointer without a cast [enabled by default]
   __pyx_t_2 = __Pyx_PyInt_From_uint64_t(__pyx_v_statbuf.f_fsid); if (unlikely(!__pyx_t_2)) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 409; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
   ^
/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/rgw/rgw_file.c:770:32: note: expected ‘uint64_t’ but argument is of type ‘uint64_t *’
 static CYTHON_INLINE PyObject* __Pyx_PyInt_From_uint64_t(uint64_t value);
                                ^
/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/rgw/rgw_file.c:3245:56: error: ‘struct rgw_statvfs’ has no member named ‘f_fsid2’
   __pyx_t_4 = __Pyx_PyInt_From_uint64_t(__pyx_v_statbuf.f_fsid2); if (unlikely(!__pyx_t_4)) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 409; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
                                                        ^
error: command '/usr/bin/cc' failed with exit status 1

**Comment by tchaikov** [comment]
@yuyuyu101 i think #11825 is a better fix. what do you think?

**Comment by dang** [comment]
I think the correct fix is this: https://github.com/ceph/ceph/pull/11831

**Pull request closed without merging**